go/vt/vtctl/grpcvtctldserver/server_test.go

```diff
@@ -8914,3 +8915,681 @@ func TestValidateShard(t *testing.T) {
+			schemaMap: map[string]*tabletmanagerdatapb.SchemaDefinition{
+				"-80": {
+					TableDefinitions: []*tabletmanagerdatapb.TableDefinition{
+						{
+							Name:    "t1",
+							Columns: []string{"c1"},
+						},
+						{
+							Name:    "t2",
+							Columns: []string{"c1"},
+						},
+					},
+				},
+				"80-": {
+					TableDefinitions: []*tabletmanagerdatapb.TableDefinition{
+						{
+							Name:    "t1",
+							Columns: []string{"c1"},
+						},
+						{
+							Name:    "t2",
+							Columns: []string{"c1"},
+						},
+					},
+				},
+			},
+			vschema: &vschemapb.Keyspace{
+				Sharded: true,
+				Tables: map[string]*vschemapb.Table{
+					"t1": {
+						Columns: []*vschemapb.Column{
+							{
+								Name: "c1",
+								Type: querypb.Type_BINARY,
+							},
+						},
+						ColumnVindexes: []*vschemapb.ColumnVindex{
+							{
+								Column: "c1",
+								Name:   "c1",
+							},
+						},
+					},
+				},
+				Vindexes: map[string]*vschemapb.Vindex{
+					"c1": {
+						Type: "hash",
+					},
+				},
+			},
+			cell: "zone1",
+			tmc: &testutil.TabletManagerClient{
+				GetSchemaResults: map[string]struct {
+					Schema *tabletmanagerdatapb.SchemaDefinition
+					Error  error
+				}{},
+			},
+			ts: memorytopo.NewServer("zone1"),
+			req: &vtctldatapb.ValidateSchemaKeyspaceRequest{
+				Keyspace:       "ks",
+				ExcludeTables:  nil,
+				IncludeVSchema: false,
+				IncludeViews:   false,
+				SkipNoPrimary:  false,
+			},
+			shouldErr: false,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			// Test Setup
+			tt.tmc.TopoServer = tt.ts
+
+			vtctld := testutil.NewVtctldServerWithTabletManagerClient(t, tt.ts, tt.tmc, func(ts *topo.Server) vtctlservicepb.VtctldServer {
+				return NewVtctldServer(ts)
+			})
+
+			// Create Keyspace
+			testutil.AddKeyspace(ctx, t, tt.ts, &vtctldatapb.Keyspace{
+				Name: tt.req.Keyspace,
+				Keyspace: &topodatapb.Keyspace{
+					KeyspaceType: topodatapb.KeyspaceType_NORMAL,
+				},
+			})
+
+			// Create Shard
+			var tabletID uint32 = 100
+
+			for shard, schema := range tt.schemaMap {
+
+				testutil.AddShards(ctx, t, tt.ts, &vtctldatapb.Shard{
+					Keyspace: tt.req.Keyspace,
+					Name:     shard,
+					Shard:    &topodatapb.Shard{},
+				})
+
+				primaryTablet := &topodatapb.Tablet{
+					Alias: &topodatapb.TabletAlias{
+						Cell: tt.cell,
+						Uid:  tabletID,
+					},
+					Type:      topodatapb.TabletType_PRIMARY,
+					Keyspace:  tt.req.Keyspace,
+					Shard:     shard,
+					MysqlPort: 3306,
+				}
+
+				testutil.AddTablet(ctx, t, tt.ts, primaryTablet, &testutil.AddTabletOptions{AlsoSetShardPrimary: true})
+
+				tabletID += 10
+
+				tt.tmc.GetSchemaResults[topoproto.TabletAliasString(primaryTablet.Alias)] = struct {
+					Schema *tabletmanagerdatapb.SchemaDefinition
+					Error  error
+				}{
+					Schema: schema,
+				}
+			}
+
+			// ApplyVSchema
+			req3 := &vtctldatapb.ApplyVSchemaRequest{
```

Comment:
in general i would suggest `applyVSchemaReq`, and also after the refactors you did, req1 and req2 appear to be gone, so this doubly needs a rename